src/main/java/games/strategy/engine/framework/ui/NewGameChooser.java

```diff
@@ -193,9 +198,22 @@ public void mouseExited(final MouseEvent e) {}
     });
   }
 
-  /** Populates the NewGameChooserModel cache if empty, then returns the cached instance. */
+  /**
+   * Populates the NewGameChooserModel cache if empty, then returns the cached instance.
+   * 
+   * @param doneAction A Runnable being executed on the EDT after the NewGameChooserModel has
+   *        been instantiated.
+   */
+  public static synchronized NewGameChooserModel getNewGameChooserModel(final Runnable doneAction) {
```

Comment:
It's not clear why this static method (as well as the `getNewGameChooserModel()` overload below) needs to be synchronized.  I don't see any static state in this class that is modified as a result of these calls.

Because of the constraint that a `NewGameChooserModel` must be constructed on the EDT, these methods should never be called from a thread other than the EDT anyway.  So, if only a single thread will ever call these methods, the synchronization isn't necessary.

Reply:
I think the original intention was to avoid 2 threads constructing 2 model instances at the same time because of the underlying IO operations, but java should really be able to handle 2 threads reading the same file at the same time...

Reply:
And after reading the javadoc, it seems that the model used to cache it's requests, so that the directory isn't parsed every time the user clicks on that button. Removing this method in favour of the normal constructor

Reply:
It seems https://github.com/triplea-game/triplea/commit/a8ee6c81ade5917de357ff83f50bf0e26dfdd63a#diff-c8ac1e7aaa575bf40e562d87db225f85L209 removed this functionality, which made some method calls useless. I'm removing those calls instead of fixing them, simply because for one they work already (and removing them gives a performance boost), and most importantly changing this behavior would require passing instances of GameChooser to retrieve the currently active model.
We should make it a hidden option (for MapMakers), to either cache a single instance for the lifetime of TripleA, or to reload it every time the user clicks on this button, or what would be even better, have a dynamic system which updates the model every time the folder changes. (Using some of the nio APIs), but out-of-scope for this PR